crates/common/tedge_utils/src/fs_notify.rs

```diff
@@ -324,6 +257,43 @@ impl NotifyStream {
             }
         }
     }
+
+    fn get_full_path_and_mask(
+        event: &Event<OsString>,
+        files: Vec<EventDescription>,
+    ) -> Result<String, NotifyStreamError> {
```

Comment:
I have many comments on this return type.

* The empty string `""` is used to mean nothing. => The return type should wrap an `Option<String>` instead of a `<String>`.
* I also wonder how this function can fail. It happens that this is only when the event is unexpected. i.e no path matches the event. => Return `None` when there is no match and the `Result` type is then useless.
* What you need at the end is a `PathBuf` not a `String`. => Return a `PathBuf`.

```suggestion
    ) -> Option<PathBuf> {
```

Reply:
Thanks, changed it to `Option<PathBuf>`